From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: OpenBMC <openbmc@lists.ozlabs.org>
Date: Mon, 25 Aug 2026 12:00:00 +0000
Subject: [PATCH] ARM: dts: aspeed: Add Supermicro X10SRL-F

Add device tree for Supermicro X10SRL-F server board with
AST2400 BMC.

Signed-off-by: OpenBMC <openbmc@lists.ozlabs.org>
---
 arch/arm/boot/dts/aspeed/Makefile                  |   1 +
 .../aspeed/aspeed-bmc-supermicro-x10srl.dts        | 271 +++++++++++++++++++++
 2 files changed, 272 insertions(+)
 create mode 100644 arch/arm/boot/dts/aspeed/aspeed-bmc-supermicro-x10srl.dts

diff --git a/arch/arm/boot/dts/aspeed/Makefile b/arch/arm/boot/dts/aspeed/Makefile
index 000000000000..111111111111 100644
--- a/arch/arm/boot/dts/aspeed/Makefile
+++ b/arch/arm/boot/dts/aspeed/Makefile
@@ -1,5 +1,6 @@
 # SPDX-License-Identifier: GPL-2.0
 dtb-$(CONFIG_ARCH_ASPEED) += \
+	aspeed-bmc-supermicro-x10srl.dtb \
 	aspeed-ast2500-evb.dtb \
 	aspeed-ast2600-evb.dtb \
 	aspeed-bmc-amd-ethanolx.dtb \
diff --git a/arch/arm/boot/dts/aspeed/aspeed-bmc-supermicro-x10srl.dts b/arch/arm/boot/dts/aspeed/aspeed-bmc-supermicro-x10srl.dts
new file mode 100644
index 000000000000..222222222222
--- /dev/null
+++ b/arch/arm/boot/dts/aspeed/aspeed-bmc-supermicro-x10srl.dts
@@ -0,0 +1,271 @@
+// SPDX-License-Identifier: GPL-2.0+
+/dts-v1/;
+
+#include "aspeed-g4.dtsi"
+#include <dt-bindings/gpio/aspeed-gpio.h>
+
+/ {
+    model = "Supermicro X10SRL-F BMC";
+    compatible = "supermicro,x10srl-bmc", "aspeed,ast2400";
+
+    chosen {
+        stdout-path = &uart5;
+        bootargs = "console=ttyS4,115200 earlyprintk";
+    };
+
+    memory@40000000 {
+        reg = <0x40000000 0x20000000>; // 512MB
+    };
+
+    reserved-memory {
+        #address-cells = <1>;
+        #size-cells = <1>;
+        ranges;
+
+        vga_memory: framebuffer@5f000000 {
+            no-map;
+            reg = <0x5f000000 0x01000000>; // 16MB
+        };
+
+        flash_memory: region@5ee00000 {
+            no-map;
+            reg = <0x5ee00000 0x00200000>; // 2MB for FW
+        };
+    };
+
+    leds {
+        compatible = "gpio-leds";
+
+        led-identify {
+            label = "identify";
+            gpios = <&gpio ASPEED_GPIO(A, 0) GPIO_ACTIVE_LOW>;
+        };
+
+        led-status {
+            label = "status";
+            gpios = <&gpio ASPEED_GPIO(A, 1) GPIO_ACTIVE_LOW>;
+        };
+    };
+
+    iio-hwmon {
+        compatible = "iio-hwmon";
+        io-channels = <&adc 0>, <&adc 1>, <&adc 2>, <&adc 3>,
+                      <&adc 4>, <&adc 5>, <&adc 6>, <&adc 7>;
+    };
+
+    iio-hwmon-battery {
+        compatible = "iio-hwmon";
+        io-channels = <&adc 8>;
+    };
+};
+
+&fmc {
+    status = "okay";
+    flash@0 {
+        status = "okay";
+        m25p,fast-read;
+        label = "bmc";
+        spi-max-frequency = <50000000>;
+#include "openbmc-flash-layout.dtsi"
+    };
+};
+
+&spi {
+    status = "okay";
+    pinctrl-names = "default";
+    pinctrl-0 = <&pinctrl_spi1_default>;
+
+    flash@0 {
+        status = "okay";
+        m25p,fast-read;
+        label = "pnor";
+        spi-max-frequency = <100000000>;
+    };
+};
+
+&uart5 {
+    status = "okay";
+};
+
+&mac0 {
+    status = "okay";
+    use-ncsi;
+    pinctrl-names = "default";
+    pinctrl-0 = <&pinctrl_rmii1_default>;
+};
+
+&mac1 {
+    status = "okay";
+    pinctrl-names = "default";
+    pinctrl-0 = <&pinctrl_rgmii2_default>;
+};
+
+&i2c0 {
+    status = "okay";
+    // FRU EEPROM
+    eeprom@50 {
+        compatible = "atmel,24c256";
+        reg = <0x50>;
+        pagesize = <64>;
+    };
+};
+
+&i2c1 {
+    status = "okay";
+    // System management bus
+};
+
+&i2c2 {
+    status = "okay";
+    // PCIe slot I2C
+};
+
+&i2c3 {
+    status = "okay";
+    // Reserved
+};
+
+&i2c4 {
+    status = "okay";
+    // Reserved
+};
+
+&i2c5 {
+    status = "okay";
+    // Reserved
+};
+
+&i2c6 {
+    status = "okay";
+    // VR/Voltage regulators
+};
+
+&i2c7 {
+    status = "okay";
+    // Thermal sensors
+    lm75@48 {
+        compatible = "national,lm75";
+        reg = <0x48>;
+    };
+
+    lm75@49 {
+        compatible = "national,lm75";
+        reg = <0x49>;
+    };
+};
+
+&i2c8 {
+    status = "okay";
+    // Reserved
+};
+
+&i2c9 {
+    status = "okay";
+    // Reserved
+};
+
+&i2c10 {
+    status = "okay";
+    // Reserved
+};
+
+&i2c11 {
+    status = "okay";
+    // Reserved
+};
+
+&i2c12 {
+    status = "okay";
+    // Reserved
+};
+
+&i2c13 {
+    status = "okay";
+    // Reserved
+};
+
+&gpio {
+    status = "okay";
+    gpio-line-names =
+    /*A0-A7*/   "LED_IDENTIFY","LED_STATUS","","","","","","",
+    /*B0-B7*/   "","","","","","","","",
+    /*C0-C7*/   "","","","","","","","",
+    /*D0-D7*/   "","","","","","","","",
+    /*E0-E7*/   "","","","","","","","",
+    /*F0-F7*/   "","","","","","","","",
+    /*G0-G7*/   "","","","","","","","",
+    /*H0-H7*/   "","","","","","","","",
+    /*I0-I7*/   "","","","","","","","",
+    /*J0-J7*/   "","","","","","","","",
+    /*K0-K7*/   "","","","","","","","",
+    /*L0-L7*/   "","","","","","","","",
+    /*M0-M7*/   "","","","","","","","",
+    /*N0-N7*/   "","","","","","","","",
+    /*O0-O7*/   "","","","","","","","",
+    /*P0-P7*/   "","","","","","","","",
+    /*Q0-Q7*/   "","","","","","","","",
+    /*R0-R7*/   "","","","","","","","",
+    /*S0-S7*/   "","","","","","","","",
+    /*T0-T7*/   "","","","","","","","",
+    /*U0-U7*/   "","","","","","","","",
+    /*V0-V7*/   "","","","","","","","",
+    /*W0-W7*/   "","","","","","","","",
+    /*X0-X7*/   "","","","","","","","",
+    /*Y0-Y7*/   "","","","","","","","",
+    /*Z0-Z7*/   "","","","","","","","",
+    /*AA0-AA7*/ "","","","","","","","",
+    /*AB0-AB7*/ "","","","","","","","";
+};
+
+&adc {
+    status = "okay";
+};
+
+&pwm_tacho {
+    status = "okay";
+    pinctrl-names = "default";
+    pinctrl-0 = <&pinctrl_pwm0_default &pinctrl_pwm1_default
+                 &pinctrl_pwm2_default &pinctrl_pwm3_default
+                 &pinctrl_pwm4_default &pinctrl_pwm5_default
+                 &pinctrl_pwm6_default &pinctrl_pwm7_default>;
+
+    fan@0 {
+        reg = <0x00>;
+        aspeed,fan-tach-ch = /bits/ 8 <0x00>;
+    };
+
+    fan@1 {
+        reg = <0x01>;
+        aspeed,fan-tach-ch = /bits/ 8 <0x01>;
+    };
+
+    fan@2 {
+        reg = <0x02>;
+        aspeed,fan-tach-ch = /bits/ 8 <0x02>;
+    };
+
+    fan@3 {
+        reg = <0x03>;
+        aspeed,fan-tach-ch = /bits/ 8 <0x03>;
+    };
+
+    fan@4 {
+        reg = <0x04>;
+        aspeed,fan-tach-ch = /bits/ 8 <0x04>;
+    };
+};
+
+&kcs3 {
+    status = "okay";
+    aspeed,lpc-io-reg = <0xca2>;
+};
+
+&lpc_ctrl {
+    status = "okay";
+};
+
+&lpc_snoop {
+    status = "okay";
+    snoop-ports = <0x80>;
+};
+--
+2.45.2
+
